app/src/main/java/edu/ucandroid/weathernotice/service/WeatherService.kt

```diff
@@ -37,17 +37,17 @@ class WeatherService {
                     call: Call<ArrayList<Weather>>,
                     response: Response<ArrayList<Weather>>
             ) {
-                _weather.value = response.body()
+                weather.value = response.body()
             }
 
         })
-        return _weather
+        return weather
     }
 
-    fun fetchWeather(locationName: String) : MutableLiveData<ArrayList<Weather>> {
-        var _weather = MutableLiveData<ArrayList<Weather>>()
+    fun fetchWeatherByLocation(locationName: String) : MutableLiveData<ArrayList<Weather>> {
+        val weather = MutableLiveData<ArrayList<Weather>>()
         val service = RetrofitClientInstance.retrofitInstance?.create(IWeatherDAO::class.java)
-        val call = service?.getAllWeather()
+        val call = service?.getWeather(locationName)
```

Comment:
I believe this was the function you were looking for here

Reply:
not yet, due other branch's that we unable to complete.